notebooks/nessie-iceberg-demo-nba.ipynb

```diff
@@ -0,0 +1,629 @@
+    "spark.sql(\"USE REFERENCE dev IN dev_catalog\")\n",
+    "\n",
+    "# Creating `salaries` table\n",
+    "spark.sql(\"CREATE TABLE IF NOT EXISTS dev_catalog.nba.salaries (Season STRING, Team STRING, Salary STRING, Player STRING) USING iceberg\")\n",
+    "\n",
+    "spark.sql('CREATE OR REPLACE TEMPORARY VIEW salaries_table USING csv OPTIONS (path \"../datasets/nba/salaries.csv\", header true)')\n",
+    "spark.sql('INSERT INTO dev_catalog.nba.salaries SELECT * FROM salaries_table')\n",
+    "\n",
+    "# Creating `totals_stats` table\n",
+    "spark.sql(\"\"\"CREATE TABLE IF NOT EXISTS dev_catalog.nba.totals_stats (\n",
+    "             Season STRING, Age STRING, Team STRING, ORB STRING, DRB STRING, TRB STRING, AST STRING, STL STRING, \n",
+    "             BLK STRING, TOV STRING, PTS STRING, Player STRING, RSorPO STRING) \n",
+    "             USING iceberg\"\"\")\n",
+    "spark.sql('CREATE OR REPLACE TEMPORARY VIEW stats_table USING csv OPTIONS (path \"../datasets/nba/totals_stats.csv\", header true)')\n",
+    "spark.sql('INSERT INTO dev_catalog.nba.totals_stats SELECT * FROM stats_table').toPandas()\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we count the rows in our tables to ensure they are the same number as the csv files. Note we use the `table@branch` notation which overrides the context set by a `USE REFERENCE` command."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    },
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "table_count = spark.sql(\"select count(*) from dev_catalog.nba.`salaries@dev`\").toPandas().values[0][0]\n",
+    "csv_count = spark.sql(\"select count(*) from salaries_table\").toPandas().values[0][0]\n",
+    "assert table_count == csv_count\n",
+    "\n",
+    "table_count = spark.sql(\"select count(*) from dev_catalog.nba.`totals_stats@dev`\").toPandas().values[0][0]\n",
+    "csv_count = spark.sql(\"select count(*) from stats_table\").toPandas().values[0][0]\n",
+    "assert table_count == csv_count"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Check generated tables\n",
+    "----------------------------\n",
+    "Since we have been working solely on the `dev` branch, where we created 2 tables and added some data,\n",
+    "let's verify that the `main` branch was not altered by our changes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false
+    },
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "spark.sql(\"USE REFERENCE main IN dev_catalog\").toPandas()\n",
+    "try:\n",
+    "    spark.sql(\"SHOW TABLES IN dev_catalog\").toPandas()\n",
+    "except:\n",
+    "    print(\"Error in 0.9.0 of Nessie causes an exception when no tables found\")"
```

Comment:
Can you add the reason for this here?

Reply:
Its a rather complicated interaction b/w the api and the client. I will fix it in an iceberg PR soon but I don't want to burden the user w/ the details.

Reply:
For us: its when we set `hashOnRef` and nothing exists on the ref we get a 404